### PR TITLE
 - Some adaptations for gcc-4.9.2 and Cygwin

### PR DESCRIPTION
**Pull request by Juppit**
- touch makedir after make

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -548,103 +569,125 @@ $(TOOLCHAIN):
...
-	mv $(XTDLP)/$(MPFR_DIR)/mpfr-*/* $(XTDLP)/$(MPFR_DIR)
+	rm -rf $(XTDLP)/$(MPFR_DIR)/build
 
 $(XTDLP)/$(MPFR_DIR)/build: $(XTDLP)/$(MPFR_DIR)
 	mkdir -p $(XTDLP)/$(MPFR_DIR)/build
-	cd $(XTDLP)/$(MPFR_DIR)/build/; ../configure --prefix=$(XTBP)/mpfr --with-gmp=$(XTBP)/gmp --disable-shared --enable-static --build=$(BUILD_TARGET) --host=$(BUILD_TARGET)
-	make -C $(XTDLP)/$(MPFR_DIR)/build/	
+	cd $(XTDLP)/$(MPFR_DIR)/build/; ../$(CONFIGURE) --prefix=$(XTBP)/mpfr --with-gmp=$(XTBP)/gmp --disable-shared --enable-static --build=$(BUILD_TARGET) --host=$(BUILD_TARGET)
+	$(MAKE) -C $(XTDLP)/$(MPFR_DIR)/build/
+	@touch $(XTDLP)/$(MPFR_DIR)/build
```
Comment (DimmKirr) [inline Makefile:604]: Can you please explain why this is needed? 
wouldn't `mkdir -p $(XTDLP)/$(MPFR_DIR)/build` be enough?

File: Makefile [hunk 2]
```diff
@@ -3,9 +3,14 @@
 # Credits to Paul Sokolovsky (@pfalcon) for esp-open-sdk
 # Credits to Fabien Poussin (@fpoussin) for xtensa-lx106-elf build script
 #
+# (@Juppit) adaptation for gcc-4.9.2 and Cygwin
+# (@Juppit) touch makedir after make 
 
 
-VENDOR_SDK_VERSION = 1.4.0
+# for utils you need the tools pyinstaller and mcs
+BUILD_UTILS = n
```
Comment (DimmKirr) [inline Makefile:11]: I would prefer to have this enabled by default. you get mcs and pyinstaller when configuring environment via scripts from env/

**Comment by DimmKirr** [comment]
@Juppit  looks like you have a lot of nonrelated changes. Can you please pull latest and update the PR to reflect only ones that are for gcc-4.9.2?

We can also add style changes (indents, remove comments,etc) later, in a different PR. 

For me it sounds like it should be 2 things to add:
- Change branch on the submodule
- Apply patch

Or did I miss anything?

File: Makefile [hunk 3]
```diff
@@ -548,103 +569,125 @@ $(TOOLCHAIN):
 	git config --global core.autocrlf false
 	mkdir -p $(TOOLCHAIN)
 
+$(PATCH_DIR):
```
Comment (DimmKirr) [inline Makefile:572]: Should not that be a part of either pass1 or pass2 of gcc?
Also, I have other patches that should go into src/patches, so this task won't probably wokr for other patches (some prereqs will be missing). 
Thoughts?

File: Makefile (hunk 1 above)
Comment (Juppit) [inline Makefile:604]: If you do the make the first time, then you have a new build directory, 
so you have a clear dependency source older then build.
But the build directory is part the source, so after the make you do not 
know what is older, what newer.

Specially the configure process changes sources.
Another problem is, that I don't know if the timestamp of a parent 
directory (e.g. build) will always reflect the timestamp of subdirs or 
even files in subdirs.

At least, if there was any fault and you have to try the make once more 
(without make clean) then the dependency has changed or not.

After all my changes (touch) I got a make process, which will not make 
the targets again and again, like it was before.

Am 23.12.2015 um 01:39 schrieb Dmitry Kireev:

File: Makefile (hunk 3 above)
Comment (Juppit) [inline Makefile:572]: As I remember, there was this issue in pass2 only, I will check that.

Here the same in Pfalcon's:

> [INFO ]  Installing final compiler
> [ERROR]    xgcc: error: spawn: No such file or directory
> [ERROR]    /home/Peter/Pfalcon/crosstool-NG/.build/xtensa-lx106-elf/build/build-cc-final/xtensa-lx106-elf/libstdc++-v3/include/valarray:1233:1: internal compiler error: Segmentation fault
> [ERROR]    make[6]: **\* [xtensa-lx106-elf/bits/stdc++.h.gch/O2g.gch] Error 1
> [ERROR]    /home/Peter/Pfalcon/crosstool-NG/.build/xtensa-lx106-elf/build/build-cc-final/xtensa-lx106-elf/libstdc++-v3/include/bits/unordered_set.h:1294:1: internal compiler error: Segmentation fault

Greetings

Am 23.12.2015 um 01:57 schrieb Dmitry Kireev:

File: Makefile (hunk 2 above)
Comment (Juppit) [inline Makefile:11]: Yea, of course, it is only my personally preference, which should not be 
there in general.

Am 23.12.2015 um 01:41 schrieb Dmitry Kireev:

**Comment by Juppit** [comment]
The changes for 4.9.2 are more trivial.
I will try that, but in these days, time is rare! ;-)

Am 23.12.2015 um 01:52 schrieb Dmitry Kireev:

**Comment by DimmKirr** [comment]
@Juppit I incorporated your changes in #12 
Thank you very much for your contribution. 

There are binary builds available for test here https://bintray.com/kireevco/generic/xtensa-lx106-elf/view#files (v4.9.2)

Let's test it for some time and merge this PR.

**Pull request closed without merging**